### ISSUE TITLE
Unexpected error during intent recognition

### ISSUE THREAD

**Issue by IoSonoAndreaZ** [m50/ha-fallback-conversation#16]
I'm using gemini 

![Screenshot_20240625_201022](https://github.com/m50/ha-fallback-conversation/assets/45826618/718eacb0-1898-4048-8a36-b8fc0dddf25c)

**Comment by m50** [comment]
Hi, I do not have gemini set up, and would not like to have to pay for it to investigate.

Could you please enable debug logging for both the gemini agent and the fallback agent, and provide me the logs when attempting to use the gemini agent?

If using Gemini without fallback agent, I assume it works?

**Comment by IoSonoAndreaZ** [comment]
Hi, this is the log:

 ```
File "/usr/src/homeassistant/homeassistant/components/mobile_app/webhook.py", line 200, in handle_webhook
    ATTR_WEBHOOK_ENCRYPTED not in req_data
TypeError: argument of type 'NoneType' is not iterable
2024-06-29 18:12:43.668 ERROR (MainThread) [homeassistant.components.assist_pipeline.pipeline] Unexpected error during intent recognition
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/assist_pipeline/pipeline.py", line 994, in recognize_intent
    conversation_result = await conversation.async_converse(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/conversation/agent_manager.py", line 108, in async_converse
    result = await method(conversation_input)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/fallback_conversation/__init__.py", line 90, in async_process
    agent_names[agent_id],
    ~~~~~~~~~~~^^^^^^^^^^
KeyError: 'conversation.google_generative_ai_conversation'
```

yes, gemini works fine without fallback agent.

Also by the way, i'm not paying anything to use gemini, as when you register on google console for the first time, you receive a 250 euro bonus for two months usage

**Comment by m50** [comment]
Seems to be the same as  #17 and maybe even #13 ?

I have just released v1.0.4 which has something that *may* fix this, but no guarantee it will, as I cannot recreate #17. 

If you delete your Fallback conversation agent and recreate it, do you still have the same issue?

**Comment by IoSonoAndreaZ** [comment]
I manually updated from HACS (I tought it was automatic!) and now it works as intended, thank you!
